### PR TITLE
support for linux kernels < 3.9 and windows

### PR DESCRIPTION
**Pull request by jr01**
Fixes https://github.com/plietar/rust-mdns/issues/10

For reference here is some information about reuse_port https://stackoverflow.com/questions/14388706/socket-options-so-reuseaddr-and-so-reuseport-how-do-they-differ-do-they-mean-t

### CODE REVIEW

File: src/address_family.rs [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
+        builder.reuse_address(true)?;
+	#[cfg(not(windows))]
+        match builder.reuse_port(true) {
+            Ok(_) => {},
+            // On linux kernel < 3.9 reuse_port is not available. Ignore
+            // the error on all linux kernel versions. If reuse_port is mandatory for some
+            // reason the bind() call will fail later.
+            #[cfg(target_os = "linux")]
+            Err(ref e) if e.raw_os_error() == Some(libc::ENOPROTOOPT) => {},
+            Err(err) => panic!("reuse_port failed: {}", err)
```
Comment (plietar) [inline src/address_family.rs]: Just return the error in an `Err`

**Review by plietar — CHANGES_REQUESTED** [review]
Thanks,

Unfortunately the `get_if_addrs` crate is under GPL license, which makes it inadequate for this.

Until someone reimplements the required functionality under a MIT like license Windows cannot be supported.

File: src/address_family.rs (hunk 1 above)
Comment (jr01) [reply src/address_family.rs]: Thanks, will change this. I'm very new to rust...

**Comment by jr01** [comment]
Damn. Missed the GPL license.

**Comment by jr01** [comment]
@plietar - I did the changes you requested - please let me know what you'd like me to change to this PR and I'll do it!

File: src/address_family.rs [hunk 2]
```diff
@@ -1,19 +1,31 @@
 use net2::UdpBuilder;
+#[cfg(not(windows))]
 use net2::unix::UnixUdpBuilderExt;
 use std::io;
 use std::net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr, UdpSocket};
 use super::MDNS_PORT;
+#[cfg(not(windows))]
```
Comment (michaelherger) [inline src/address_family.rs:7]: this should be
```
#[cfg(target_os = "linux")]
```
as further down the page we only use libc using that condition

**Pull request closed without merging**